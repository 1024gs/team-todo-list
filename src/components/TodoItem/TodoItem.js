import React from 'react';
import './TodoItem.scss';

const TodoItem = ({item, getItemId, onToggle, onDelete}) => {
  const handleToggle = (e) => {
    onToggle(item, e.target.checked);
  };

  const handleDelete = () => {
    onDelete(item);
  };

  return (
    <span className={`TodoItem${item.isCompleted ? ' completed' : ''}`}>
      <span className="checkbox-no-label">
        <input
          type="checkbox"
          checked={!!item.isCompleted}
          id={`TodoItem-${getItemId(item)}`}
          onChange={handleToggle}
        />
        <label htmlFor={`TodoItem-${getItemId(item)}`}></label>
      </span>

      <span className="TodoItem-title truncate">{item.title}</span>
      <button className="TodoItem-delete btn-flat" onClick={handleDelete}>
        <i className="material-icons">delete</i>
      </button>
    </span>
  );
};

export default TodoItem;
