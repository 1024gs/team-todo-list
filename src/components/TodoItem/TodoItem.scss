.TodoItem {
  display: flex;
  align-items: center;
  height: 46px;
  transition: 0.25s;
  padding: 10px 20px;
}

.TodoItem-title {
  flex-grow: 1;
  cursor: pointer;
}

.TodoItem-title input[type='text'] {
  margin-bottom: 0;
  box-shadow: none !important;
}

.TodoItem.completed .TodoItem-title {
  text-decoration: line-through;
  color: #9f9f9f;
}

.TodoItem-delete {
  padding-right: 10px;
  padding-left: 10px;
  display: none;
}

.TodoItem:hover .TodoItem-delete {
  display: inline-block;
}
