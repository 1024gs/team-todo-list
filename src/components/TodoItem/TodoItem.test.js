import React from 'react';
import TodoItem from './TodoItem';
import sinon from 'sinon';
import {shallow, mount} from 'enzyme';

const setup = (propOverrides, doMount) => {
  const props = Object.assign(
    {
      item: {id: 1, title: 'Todo 1'},
      getItemId: (x) => x.id,
      onToggle: sinon.spy(),
      onDelete: sinon.spy()
    },
    propOverrides
  );
  const wrapper = (doMount ? mount : shallow)(<TodoItem {...props} />);

  return {
    props,
    wrapper
  };
};

describe('TodoItem', () => {
  it('renders', () => {
    const {wrapper} = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('has class completed', () => {
    const {wrapper} = setup({item: {id: 1, title: 'a', isCompleted: true}});
    // expect(wrapper.hasClass('completed')).toBe(true);
    /* The above doesn't work. https://github.com/airbnb/enzyme/issues/307 */
    expect(wrapper.find('.TodoItem').hasClass('completed')).toBe(true);
  });

  it('has NOT class completed', () => {
    const {wrapper} = setup({item: {id: 1, title: 'a', isCompleted: false}});
    // expect(wrapper.hasClass('completed')).toBe(false);
    /* The above doesn't work. https://github.com/airbnb/enzyme/issues/307 */
    expect(wrapper.find('.TodoItem').hasClass('completed')).toBe(false);
  });

  it('calls onToggle correctly', () => {
    const {wrapper, props} = setup({item: {id: 1, title: 'a'}}, true);
    wrapper.find('input[type="checkbox"]').simulate('change');
    expect(props.onToggle.calledWithExactly({id: 1, title: 'a'}, false)).toBe(true);
    wrapper.unmount();
  });

  it('calls onDelete correctly', () => {
    const {wrapper, props} = setup({item: {id: 1, title: 'a'}}, true);
    wrapper.find('.TodoItem-delete').simulate('click');
    expect(props.onDelete.calledWithExactly({id: 1, title: 'a'})).toBe(true);
    wrapper.unmount();
  });
});
