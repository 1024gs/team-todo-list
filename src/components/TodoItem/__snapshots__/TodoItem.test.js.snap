// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders 1`] = `
<span
  className="TodoItem"
>
  <span
    className="checkbox-no-label"
  >
    <input
      checked={false}
      id="TodoItem-1"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="TodoItem-1"
    />
  </span>
  <span
    className="TodoItem-title truncate"
  >
    Todo 1
  </span>
  <button
    className="TodoItem-delete btn-flat"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      delete
    </i>
  </button>
</span>
`;
